pre.js

```diff
@@ -29,5 +29,47 @@ var Module = {
         console.error(text);
     },
     canvas: {},
-    noInitialRun: true
+    noInitialRun: true,
+    onRuntimeInitialized: function() {
+        // Expose enum values.
+        var enums = {
+            "heif_error_code": true,
+            "heif_suberror_code": true,
+            "heif_compression_format": true,
+            "heif_chroma": true,
+            "heif_colorspace": true,
+            "heif_channel": true
+        };
+        var e;
+        for (e in enums) {
+            if (!enums.hasOwnProperty(e)) {
+                continue;
+            }
+
+            for (key in this[e]) {
+                if (!this[e].hasOwnProperty(key) ||
+                    key === "values") {
+                    continue;
+                }
+
+                libheif[key] = this[e][key];
+            }
+        }
+
+        // Expose internal C API.
+        for (key in this) {
+            if (enums.hasOwnProperty(key.slice(1)) || key.indexOf("_heif_") !== 0) {
+                continue;
+            }
+            libheif[key.slice(1)] = this[key];
+        }
+
+        // Expose embind API.
+        for (key in Module) {
```

Comment:
I alias the embind API after the C API because there are two functions that have the same name as in the C API (`heif_context_read_from_memory` and `heif_get_version`), we prefer the embind function over the C API function.